src/shared/discover-image.c

```diff
@@ -1590,6 +1590,25 @@ int image_set_limit(Image *i, uint64_t referenced_max) {
         return 0;
 }
 
+int image_set_pool_limit(ImageClass class, uint64_t referenced_max) {
```

Comment:
Maybe better to add an assertion something like the following?
```c
assert(class >= 0 && class < _IMAGE_CLASS_MAX);
```

Reply:
done